### PR TITLE
Mvi core improve

### PR DESCRIPTION
**Pull request by Hukumister**

### CODE REVIEW

File: mvi-core/src/main/java/ru/coderedwolf/mvi/core/impl/BaseStore.kt [hunk 1]
```diff
@@ -59,32 +63,35 @@ abstract class BaseViewModelStore<Action, State, Effect, ViewEvent>(
...
+            .observeOn(mainScheduler.invoke())
             .subscribe(mviView::route)
             .addTo(disposable)
 
-        stateSubject
-            .observeOn(schedulerProvider.mainThread)
+        disposable += stateSubject
+            .observeOn(mainScheduler.invoke())
             .subscribe(mviView::render)
             .addTo(disposable)
```
Comment (kdk96) [inline mvi-core/src/main/java/ru/coderedwolf/mvi/core/impl/BaseStore.kt]: wtf is going on?

File: mvi-core-impl/proguard-rules.pro [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+# Uncomment this to preserve the line number information for
+# debugging stack traces.
+#-keepattributes SourceFile,LineNumberTable
+
+# If you keep the line number information, uncomment this to
+# hide the original source file name.
```
Comment (kdk96) [inline mvi-core-impl/proguard-rules.pro]: you must delete this file

File: mvi-core-impl/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (kdk96) [inline mvi-core-impl/.gitignore]: you must delete this file, see what you add to commit

**Review by kdk96 — CHANGES_REQUESTED** [review]

File: mvi-core/src/main/java/ru/coderedwolf/mvi/core/impl/BaseStore.kt (hunk 1 above)
Comment (Hukumister) [reply mvi-core/src/main/java/ru/coderedwolf/mvi/core/impl/BaseStore.kt]: hah, i missed it

Comment (Hukumister) [reply mvi-core/src/main/java/ru/coderedwolf/mvi/core/impl/BaseStore.kt]: done

File: mvi-core-impl/proguard-rules.pro (hunk 2 above)
Comment (Hukumister) [reply mvi-core-impl/proguard-rules.pro]: done

File: mvi-core-impl/.gitignore (hunk 3 above)
Comment (Hukumister) [reply mvi-core-impl/.gitignore]: done

**Comment by Hukumister** [comment]
[Реализовать свой MVIcore](https://trello.com/c/Mb9MmXbN/30-%D1%80%D0%B5%D0%B0%D0%BB%D0%B8%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D1%8C-%D1%81%D0%B2%D0%BE%D0%B9-mvicore)

**Pull request merged**